pkg/reconciler/openshift/tektonpipeline/installerset.go

```diff
@@ -130,27 +147,23 @@ func makeInstallerSet(tp *v1alpha1.TektonPipeline, manifest mf.Manifest, prefix,
 	}
 }
 
-func deleteInstallerSet(ctx context.Context, oc clientset.Interface, ta *v1alpha1.TektonPipeline, component string) error {
+func deleteInstallerSet(ctx context.Context, oc clientset.Interface, ta *v1alpha1.TektonPipeline, component, labelSelector string) error {
 
-	compInstallerSet, ok := ta.Status.ExtentionInstallerSets[component]
-	if !ok {
-		return nil
+	// delete the installer set
+	err := oc.OperatorV1alpha1().TektonInstallerSets().
+		DeleteCollection(ctx, metav1.DeleteOptions{},
+			metav1.ListOptions{
+				LabelSelector: labelSelector,
+			})
+	if err != nil {
+		return err
 	}
 
-	if compInstallerSet != "" {
-		// delete the installer set
-		err := oc.OperatorV1alpha1().TektonInstallerSets().
-			Delete(ctx, ta.Status.ExtentionInstallerSets[component], metav1.DeleteOptions{})
-		if err != nil && !errors.IsNotFound(err) {
-			return err
-		}
-
-		// clear the name of installer set from TektonPipeline status
-		delete(ta.Status.ExtentionInstallerSets, component)
-		_, err = oc.OperatorV1alpha1().TektonPipelines().UpdateStatus(ctx, ta, metav1.UpdateOptions{})
-		if err != nil && !errors.IsNotFound(err) {
-			return err
-		}
+	// clear the name of installer set from TektonPipeline status
+	delete(ta.Status.ExtentionInstallerSets, component)
+	_, err = oc.OperatorV1alpha1().TektonPipelines().UpdateStatus(ctx, ta, metav1.UpdateOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
```

Comment:
As we are not storing installerset name in status 
do we still required to do this :thinking:

Reply:
i will capture this in an issue and remove this part properly as this will involve change to Status struct.
ie, let us make that change in a minor release instead of a patch release.

That way we can properly deprecate that field.

issue: https://github.com/tektoncd/operator/issues/590